### PR TITLE
f-button@v0.5.0 -  add navigation link button

### PR DESCRIPTION
**Pull request by FayeCarter**
### Added
- `Action` component for standard buttons.
- `Link` component for links styled as buttons.

### CODE REVIEW

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 1]
```diff
@@ -28,6 +33,10 @@ export default {
         isFullWidth: {
             type: Boolean,
             default: false
+        },
+        href: {
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue]: You shouldn't need to define href as a prop, as it's just a standard attribute (so it should be applied automatically onto the component when set).

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
 </template>
 
 <script>
+import ActionButton from './Action.vue';
+import LinkButton from './Link.vue';
 
 export default {
     name: 'FButton',
-    components: {},
+    components: { ActionButton, LinkButton },
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue]:
```suggestion
    components: {
        ActionButton,
        LinkButton
     },
```

File: packages/components/atoms/f-button/stories/Button.stories.js [hunk 3]
```diff
@@ -20,9 +20,12 @@ export const ButtonComponent = () => ({
         },
         isFullWidth: {
             default: boolean('isFullWidth', false)
+        },
+        href: {
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/stories/Button.stories.js:27]: Not sure the best way to test this in storybook, but this should just be applied as a standard attribute (may be worth looking up if it's possible to link attributes up to knobs or not).

File: packages/components/atoms/f-button/src/components/Link.vue [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<template>
+    <a
+        :href="$attrs.href"
+        target="_blank">
```
Comment (dandel10n) [inline packages/components/atoms/f-button/src/components/Link.vue]: I don't think we need to rewrite a default target prop here

**Review by ashleynolan — COMMENTED** [review]
Couple of small comments – nothing major though, looks good.

**Review by dandel10n — COMMENTED** [review]
Looks good, one comment

File: packages/components/atoms/f-button/src/components/Link.vue (hunk 4 above)
Comment (xander-marjoram) [reply packages/components/atoms/f-button/src/components/Link.vue]: Do we need `href` passed through as well?

Comment (FayeCarter) [reply packages/components/atoms/f-button/src/components/Link.vue]: 👍  I've removed both

File: packages/components/atoms/f-button/src/components/Button.vue (hunk 1 above)
Comment (FayeCarter) [reply packages/components/atoms/f-button/src/components/Button.vue]: 👍 removed, thanks

File: packages/components/atoms/f-button/src/components/_tests/Button.test.js [hunk 5]
```diff
@@ -8,4 +8,81 @@ describe('Button', () => {
...
+                // Assert
+                expect(wrapper.vm.buttonSizeClassname).toEqual('XSmall');
+            });
+        });
+
+        describe('componentType :: ', () => {
+            describe('when `href` prop is an empty string :: :: ', () => {
+                const propsData = { href: '' };
+
+                it('should return `action`', () => {
```
Comment (kevinrodrigues) [inline packages/components/atoms/f-button/src/components/_tests/Button.test.js]: `Action`

File: packages/components/atoms/f-button/src/components/_tests/Button.test.js [hunk 6]
```diff
@@ -8,4 +8,81 @@ describe('Button', () => {
...
+                    // Assert
+                    expect(wrapper.find('[data-test-id="action-button-component"]').exists()).toBeTruthy();
+                });
+            });
+
+            describe('when `href` prop is not an empty string :: :: ', () => {
+                const link = 'http://www.just-eat.co.uk';
+                const propsData = { href: link };
+
+                it('should return `link`', () => {
```
Comment (kevinrodrigues) [inline packages/components/atoms/f-button/src/components/_tests/Button.test.js]: `Link`?

File: packages/components/atoms/f-button/src/components/_tests/Button.test.js [hunk 7]
```diff
@@ -8,4 +8,81 @@ describe('Button', () => {
...
+                    // Act
+                    const wrapper = mount(FButton, { propsData });
+
+                    // Assert
+                    expect(wrapper.find('[data-test-id="action-button-component"]').exists()).toBeTruthy();
+                });
+            });
+
+            describe('when `href` prop is not an empty string :: :: ', () => {
+                const link = 'http://www.just-eat.co.uk';
```
Comment (kevinrodrigues) [inline packages/components/atoms/f-button/src/components/_tests/Button.test.js]: Should go in a `BeforeEach`

File: packages/components/atoms/f-button/src/components/_tests/Button.test.js [hunk 8]
```diff
@@ -8,4 +8,81 @@ describe('Button', () => {
...
+                const wrapper = shallowMount(FButton, { propsData });
+
+                // Assert
+                expect(wrapper.vm.buttonSizeClassname).toEqual('XSmall');
+            });
+        });
+
+        describe('componentType :: ', () => {
+            describe('when `href` prop is an empty string :: :: ', () => {
+                const propsData = { href: '' };
```
Comment (kevinrodrigues) [inline packages/components/atoms/f-button/src/components/_tests/Button.test.js]: beforeEach

File: packages/components/atoms/f-button/src/components/Action.vue [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<template>
+    <button>
```
Comment (kevinrodrigues) [inline packages/components/atoms/f-button/src/components/Action.vue]: Probably missed the change somewhere but is it possible to set this to `button=type` by default if the button component is used? To avoid setting it at component level.

Comment (FayeCarter) [reply packages/components/atoms/f-button/src/components/Action.vue]: 👍  thanks

File: packages/components/atoms/f-button/src/components/_tests/Button.test.js (hunk 6 above)
Comment (FayeCarter) [reply packages/components/atoms/f-button/src/components/_tests/Button.test.js]: changed to `componentType` of `link-button` after chat :)

File: packages/components/atoms/f-button/src/components/_tests/Button.test.js (hunk 5 above)
Comment (FayeCarter) [reply packages/components/atoms/f-button/src/components/_tests/Button.test.js]: changed to `componentType` of `action-button` after chat :)

File: packages/components/atoms/f-button/src/components/Action.vue [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<template>
+    <button type="button">
```
Comment (kevinrodrigues) [inline packages/components/atoms/f-button/src/components/Action.vue]: You probably want to set this dynamically? You could pass through an option to allow attributes to be passed through and then you could dynamically set them via a directive? Maybe make use of the prop on the button.vue component that exposes the $attrs already? Happy for this to be looked in another pr btw.

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/components/atoms/f-button/src/components/Action.vue (hunk 10 above)
Comment (FayeCarter) [reply packages/components/atoms/f-button/src/components/Action.vue]: @kevinrodrigues I _think_ I've got this working now 🤞

**Review by kevinrodrigues — APPROVED** [review]

**Review by simonsandersje — APPROVED** [review]
